docs/blocks-storage/compactor.md

```diff
@@ -295,6 +295,16 @@ compactor:
   # CLI flag: -compactor.block-visit-marker-file-update-interval
   [block_visit_marker_file_update_interval: <duration> | default = 1m]
 
+  # How long cleaner visit marker file should be considered as expired and able
+  # to be picked up by cleaner again.
+  # CLI flag: -compactor.cleaner-visit-marker-timeout
+  [cleaner_visit_marker_timeout: <duration> | default = 10m]
```

Comment:
In theory, cleaner visit marker should not introduce any behavioural difference. Because cleaner is still using ring to assign tenants to one cleaner. In this case, no more than one cleaner would compete cleaning for same tenant. However, visit marker would be useful when end user trying to migrate from existing compactor deployment to new one while two sets of deployments would be existing at the same for period of time.

Reply:
I'm agreeing. This feature shouldn't be very disruptive. We can avoid creating yet another feature flag.